dev-embedded/arduino: bump to 1.8.5

hsoft opened the pull request:
This is a major rewrite for a major bump. As we can see in the bug
referred below, a *lot* of work can be poured into this ebuild but we
can also see that having too large a scope makes us never bump. Thus, I
opted for a bump that didn't widen the scope of the v1.0.5 ebuild. We
can always improve upon this ebuild later.

This work is strongly inspired by maurerpe's fine work at
https://github.com/maurerpe/arduino-overlay/tree/master/dev-embedded/arduino-ide
but I scaled down the unbundling effort for simplicity's sake.

Bug: https://bugs.gentoo.org/525882
Package-Manager: Portage-2.3.19, Repoman-2.3.6


`dev-embedded/arduino-ctags` is needed because it's arduino's private fork and *has* to be used for things to work.

`dev-embedded/arduino-listserialportsc` is a separate package to avoid QA warnings due to `.so` files being installed while `RESTRICT="bincheck"` was on.

`dev-embedded/arduino-builder` is unbundled because it's a good first effort on arduino's part to separate its IDE from its build system and is relatively simple to build on its own.

hsoft commented:
ping @tamiko

hsoft commented:
Maybe relevant: previous attempt at #5918

hsoft commented:
Also, depends on #7160

hsoft commented on dev-embedded/arduino-ctags/arduino-ctags-20161123.ebuild:
```diff
@@ -0,0 +1,30 @@
+
+COMMITHASH="abc8fca7499f44c725122881cd380a88c37abe0e"
+
+DESCRIPTION="Arduino private fork of ctags"
```
@jonasstein I'm glad you touch this subject, I was wondering about that. I completely agree that this package needs a better description but it involves a contextualization that can be a bit verbose. Where should that context go, in bash comments right over the `DESCRIPTION` assignment? at the top of the file? I didn't see many ebuilds that are verbosely commented.

hsoft commented:
@jonasstein thanks for your review, will fix soon.

hsoft commented:
@jonasstein I did run `repoman full` (without the `-x` first, but now I've just ran it with `-x`) but I get no warning that isn't about the old arduino 1.0.5 ebuild. You get warnings on your side? Do you think there's something misconfigured on my machine?

jonasstein commented:
please try to merge all comments and do not create a new one like in a chat. I deleted and merged some. 
I did not use repoman, because I saw so many errors with my eye already. I expected repoman to cry.

hsoft commented on dev-embedded/arduino-builder/arduino-builder-1.3.25.ebuild:
```diff
@@ -0,0 +1,79 @@
+}
+
+pkg_postinst() {
+	[ ! -x /usr/bin/avr-gcc ] && ewarn "Missing avr-gcc; you need to crossdev -s4 avr"
```
I don't know either. I inherited this from the previous package. Since I don't know, I prefer to be prudent and leave it as-is...

jonasstein replied:
No, better ask on gentoo-dev-help. If you see something strange, do not look away. I do not know anybody, who knows everything.

hsoft replied:
I'll try, but I somehow have the impression that I'll hear crickets on that kind of question...

Amynka replied:
It is standard way to do that don't worry about it.

hsoft commented:
I think I've addressed all remarks from @jonasstein in these new commits I've just rebased, squashed and pushed.

I ran `repoman full -x` on all 4 packages and the only warnings I got were for the arduino-1.0.5* ebuilds.

jonasstein commented on dev-embedded/arduino-builder/metadata.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
```
what about <longdescription>?

hsoft replied:
I added it to `arduino-ctags` which needed context but I wasn't under the impression that it was needed for `arduino-builder`. The `DESCRIPTION` seems clear enough to me.
